torch/csrc/distributed/c10d/Store.cpp

```diff
@@ -38,18 +38,34 @@ std::string Store::get_to_str(const std::string& key) {
 }
 
 void Store::append(const std::string& key, const std::vector<uint8_t>& value) {
-  TORCH_CHECK(false, "Not implemented.");
+  // This fallback depends on compareSet
+  std::vector<uint8_t> expected = value;
+  std::vector<uint8_t> current;
+  // cannot use get(key) as it might block forever if the key doesn't exist
+  current = compareSet(key, current, expected);
+  while (current != expected) {
+    expected = current;
+    expected.insert(expected.end(), value.begin(), value.end());
+    current = compareSet(key, current, expected);
+  }
```

Comment:
How can we prevent forever loop?

Reply:
In theory, we can't prevent infinite loops when using CAS as long as there are multiple active clients going at each other.

OTOH, we don't have spurious, no progress, CAS failures so each time it fails there has being global progress.